### PR TITLE
[Merged by Bors] - Avoid Printing Binary String to Logs

### PR DESCRIPTION
**Pull request by darcys22**
Converts the graffiti binary data to string before printing to logs.

## Issue Addressed

#1566 

## Proposed Changes
Rather than converting graffiti to a vector the binary data less the last character is passed to String::from_utf_lossy(). This then allows us to call the to_string() function directly to give us the string

## Additional Info

Rust skills are fairly weak

### CODE REVIEW

**Comment by AgeManning** [comment]
Thanks for the contribution! 

I've commented on the original issue as I think there is more required to resolve this. 

I'll let @michaelsproul comment on what we want to do here.

**Comment by darcys22** [comment]
Updated pull request adding regex to search through the bytes for invisible control characters and unused code points. 

Appears to be working, logs no longer have unprintable characters:

seandarcy_—_sean____dev____119_77_85_139_—_ssh_sean_192_168_1_98_—_Solarized_Dark_xterm-256color_—_120×33

looking at the hex on the block explorer:
https://beaconscan.com/slot/193319

The hex for the graffiti is `Hex:0x4c69676874686f7573652f76302e322e382d3936373730306300000000000000` which previously was creating characters that could not be displayed from the zeros

**Comment by darcys22** [comment]
Could I request some assistance with this failing lint test:
https://rust-lang.github.io/rust-clippy/master/index.html#invalid_regex

`Regex::new(r"\p{C}")` causes `regex syntax error: Unicode not allowed here`

I think because `Regex::new("\p{C}")` fails the linter believes that`Regex::new(r"\p{C}")` also fails

File: beacon_node/beacon_chain/src/beacon_chain.rs [hunk 1]
```diff
@@ -1319,8 +1320,10 @@ impl<T: BeaconChainTypes> BeaconChain<T> {
         block: SignedBeaconBlock<T::EthSpec>,
     ) -> Result<GossipVerifiedBlock<T>, BlockError<T::EthSpec>> {
         let slot = block.message.slot;
-        let graffiti_string = String::from_utf8(block.message.body.graffiti[..].to_vec())
-            .unwrap_or_else(|_| format!("{:?}", &block.message.body.graffiti[..]));
+        let re = Regex::new(r"\p{C}").unwrap();
```
Comment (michaelsproul) [inline beacon_node/beacon_chain/src/beacon_chain.rs:1323]: To work around the Clippy issue we can write `"\\p{C}"` here instead. I opened https://github.com/rust-lang/rust-clippy/issues/6005 to track the false positive.

The `unwrap` would maybe be better as an `.expect("regex is valid")`

File: beacon_node/beacon_chain/src/beacon_chain.rs [hunk 2]
```diff
@@ -1319,8 +1320,10 @@ impl<T: BeaconChainTypes> BeaconChain<T> {
         block: SignedBeaconBlock<T::EthSpec>,
     ) -> Result<GossipVerifiedBlock<T>, BlockError<T::EthSpec>> {
         let slot = block.message.slot;
-        let graffiti_string = String::from_utf8(block.message.body.graffiti[..].to_vec())
-            .unwrap_or_else(|_| format!("{:?}", &block.message.body.graffiti[..]));
+        let re = Regex::new("\\p{C}").expect("regex is valid");
```
Comment (michaelsproul) [inline beacon_node/beacon_chain/src/beacon_chain.rs:1323]:
```suggestion
        #[allow(clippy::invalid_regex)]
        let re = Regex::new("\\p{C}").expect("regex is valid");
```

Comment (michaelsproul) [reply beacon_node/beacon_chain/src/beacon_chain.rs:1323]: It seems I was wrong about the raw string being part of the problem. Clippy is really wrong here, so I think we should just bypass it.

Comment (darcys22) [reply beacon_node/beacon_chain/src/beacon_chain.rs:1324]: Frustrating, appreciate the review!

**Comment by michaelsproul** [comment]
The macOS tests are flaky, I'm going to try merging

bors r+

**Review by michaelsproul — APPROVED** [review]

**Pull request closed without merging**